install/debian/postinst.in

```diff
@@ -1,59 +1,64 @@
 #!/bin/sh -e
 
 NAME=fah-client
+CLIENT_CONFIG=/etc/$NAME
 CLIENT_HOME=/var/lib/$NAME
 CLIENT_LOGS=/var/log/$NAME
 
-case "$1" in
-  reconfigure|configure)
-    # Create group if it does not exist
-    if ! getent group $NAME >/dev/null; then
-      groupadd --force --system $NAME
-    fi
-
-    # Create user if it does not exist
-    if ! getent passwd $NAME >/dev/null; then
-      useradd --system --gid $NAME --shell /usr/sbin/nologin \
-        --home-dir $CLIENT_HOME --no-create-home --groups video,render $NAME
-    fi
-
-    # Create directories
-    install -d -m 755 -o $NAME -g $NAME $CLIENT_HOME $CLIENT_LOGS
-
-    # Create a symbolic link to the configuration file
-    ln -sf /etc/fahclient/config.xml $CLIENT_HOME/config.xml
-
-    # Remove old unit if it exists
-    rm -f /etc/systemd/system/${NAME}.service
-
-    # Start the service
-    if [ -n "$2" ]; then
-      ACTION=restart
-    else
-      ACTION=start
-    fi
-    systemctl daemon-reload
-    systemctl enable $NAME
-    systemctl $ACTION $NAME
+# Create directories
+install -d -m 755 -o $NAME -g $NAME $CLIENT_HOME $CLIENT_LOGS
 
+# migrate configuration
+
+# remove obsolete symlink installed by previous v8 packages
+[ -L $CLIENT_HOME/config.xml ] && rm -f $CLIENT_HOME/config.xml
+
+if [ ! -e $CLIENT_CONFIG/config.xml ]; then
+  if [ -f /etc/fahclient/config.xml ]; then
+    cp --remove-destination /etc/fahclient/config.xml $CLIENT_CONFIG/
     echo
-    echo "The Folding@home client is now installed"
-    echo
-    echo "File locations:"
-    echo
-    echo "  Logs: /var/log/$NAME"
-    echo "  Data: /var/lib/$NAME"
-    echo
-    echo "Service commands:"
-    echo
-    echo "  systemctl status --no-pager -l $NAME"
-    echo "  sudo systemctl start $NAME"
-    echo "  sudo systemctl stop $NAME"
-    echo "  sudo systemctl restart $NAME"
+    echo "v7 client configuration copied to $CLIENT_CONFIG/config.xml"
     echo
-    echo "Access the web interface by going to:"
+    echo "It's recommended to purge the old package to remove leftover files:"
     echo
-    echo "  %(PACKAGE_URL)s"
+    echo "  sudo apt purge fahclient"
+  elif [ -f $CLIENT_HOME/config.xml ]; then
+    mv -f $CLIENT_HOME/config.xml $CLIENT_CONFIG/
+  fi
+
+  if [ ! -e $CLIENT_CONFIG/config.xml ]; then
```

Comment:
This is unnecessary. We are already on the block from `[ ! -e $CLIENT_CONFIG/config.xml ]` above. Just put an `else` here like I wrote before: if `/etc/fah-client/config.xml` was not migrated from `/etc/fahclient/config.xml` nor from `/var/lib/fah-client/config.xml`, it does not exist.

Reply:
Ah, ok. Guess I'm easily confused.